docs/setup/independent/high-availability.md

```diff
@@ -0,0 +1,480 @@
+1. 1GB or more of RAM per machine (any less will leave little room for your apps).
+1. Full network connectivity between all machines in the cluster (public or
+   private network is fine).
+
+{% endcapture %}
+
+{% capture steps %}
+
+## Set up masters
+
+1. For each master that has been provisioned, follow the [installation guide](/docs/setup/independent/install-kubeadm/) on how to install kubeadm and its dependencies. At the end of this step, you should have all the dependencies installed on each master.
+
+## Set up etcd
+
+For highly available set ups, you will need to decide how to host your etcd cluster. A cluster is composed of at least 3 members. We recommend one of the following models:
+
+1. Hosting etcd cluster on separate compute nodes (Virtual Machines), or
+2. Hosting etcd cluster on the master nodes.
+
+Whilst the first option provides more performance and better hardware isolation, it is also more expensive and requires an additional support burden.
+
+For **Option 1**: you will need to create 3 Virtual Machines that follow [CoreOS's hardware recommendations](https://coreos.com/etcd/docs/latest/op-guide/hardware.html).
+   For the sake of simplicity, we will refer to them as `etcd0`, `etcd1` and `etcd2`.
+
+For **Option 2**: you can skip to the next step. Any reference to `etcd0`, `etcd1` and `etcd2` throughout this guide should be replaced with `master0`, `master1` and `master2` accordingly, since your master nodes host etcd.
+
+## Create etcd CA certs
+
+1. SSH into `etcd0` and run the following:
+
+    ```
+    mkdir -p /etc/kubernetes/pki/etcd
+    cd /etc/kubernetes/pki/etcd
+    ```
+    ```
+    cat >ca-config.json <<EOL
+    {
+        "signing": {
+            "default": {
+                "expiry": "43800h"
+            },
+            "profiles": {
+                "server": {
+                    "expiry": "43800h",
+                    "usages": [
+                        "signing",
+                        "key encipherment",
+                        "server auth",
+                        "client auth"
+                    ]
+                },
+                "client": {
+                    "expiry": "43800h",
+                    "usages": [
+                        "signing",
+                        "key encipherment",
+                        "client auth"
+                    ]
+                },
+                "peer": {
+                    "expiry": "43800h",
+                    "usages": [
+                        "signing",
+                        "key encipherment",
+                        "server auth",
+                        "client auth"
+                    ]
+                }
+            }
+        }
+    }
+    EOL
+    ```
+    ```
+    cat >ca-csr.json <<EOL
+    {
+        "CN": "etcd",
+        "key": {
+            "algo": "rsa",
+            "size": 2048
+        },
+        "names": [
```

Comment:
is this strictly necessary information to provide for cfssl to work?

Reply:
nope, just tried and I think this works. So I'm gonna remove this ✨